torch/linalg/__init__.py

```diff
@@ -344,49 +381,78 @@
 eigh = _add_docstr(_linalg.linalg_eigh, r"""
 linalg.eigh(input, UPLO='L', *, out=None) -> (Tensor, Tensor)
 
-Computes the eigenvalues and eigenvectors of a complex Hermitian (or real symmetric)
-matrix :attr:`input`, or of each such matrix in a batched :attr:`input`.
+Computes the eigenvalue decomposition of a complex Hermitian (or real symmetric) matrix or batch of matrices.
+
+For a complex Hermitian (or real symmetric) matrix :math:`A`, this is defined as
 
-For a single matrix :attr:`input`, the tensor of eigenvalues `w` and the tensor of eigenvectors
-`V` decompose the :attr:`input` such that `input = V diag(w) Vᴴ`, where `Vᴴ` is the transpose of `V`
-for real-valued :attr:`input`, or the conjugate transpose of `V` for complex-valued :attr:`input`.
+.. math::
+
+    A = V \operatorname{diag}(L) V^{\text{H}}
+
+where :math:`V^{\text{H}}` denotes the conjugate transpose in the complex case and the transpose in
+the real case. The matris :math:`V` (and thus :math:`V^{\text{H}}`)
+is orthogonal in the real case, and unitary in the complex case.
+
+The decomposition is represented as a namedtuple `(eigenvalues, eigenvectors)`, which
+corresepond to `(L, V)` above. If :attr:`input` is a batch of matrices, then `L`, `V` are also
+batched with the same batch dimensions.
+
+The eigenvalues are returned in ascending order.
 
-Since the matrix or matrices in :attr:`input` are assumed to be Hermitian, the imaginary part of their diagonals
-is always treated as zero. When :attr:`UPLO` is "L", its default value, only the lower triangular part of each
-matrix is used in the computation. When :attr:`UPLO` is "U" only the upper triangular part of each matrix is used.
+:attr:`input` is assumed to be Hermitian (resp. symmetric), but this is not checked internally.
+When :attr:`UPLO` is `'L'` (default), only the lower triangular part of each matrix is used in the computation.
```

Comment:
Maybe this can be documented in the arg descriptions?

Reply:
See https://github.com/pytorch/pytorch/pull/56265#discussion_r615922987